Cython/Compiler/Nodes.py

```diff
@@ -715,7 +715,8 @@ def analyse(self, return_type, env, nonempty=0, directive_locals=None, visibilit
         if (return_type.is_pyobject
                 and (self.exception_value or self.exception_check)
                 and self.exception_check != '+'):
-            error(self.pos, "Exception clause not allowed for function returning Python object")
+            # Exception clause is silently ignored for functions returning Python object.
+            self.exception_check = False
```

Comment:
Maybe we should only allow `except *` for functions returning PyObjects (which hopefully matches what gets added with the `cfunc` decorator). Thus _other_ exception statements (e.g. `except -1`) would still be an error?

Reply:
Done. I would like to implement test for this case:
```python
@cython.exceptval(-1)
@cython.cfunc
def test_cdef_return_object_broken(x: object) -> object:
    """
    >>> test_cdef_return_object_broken(3)
    # Here I should somehow assert that error() was called.
    """
    return x
```
but I am not sure how I can assert following error:
```
=== Got: ===
105:0: Exception clause not allowed for function returning Python object
```

Reply:
The files to check for compile-time errors are in tests/errors. If you follow one of them it'll show you how to do it

Reply:
Done.